### PR TITLE
Backport CreatePixmap allocation hints ABI change from X.org

### PR DESCRIPTION
**Pull request by sunweaver**
This PR requires #409 to be merged first.

### CODE REVIEW

File: nx-X11/programs/Xserver/hw/nxagent/Drawable.c [hunk 1]
```diff
@@ -2702,7 +2702,7 @@ void nxagentCreateDrawableBitmap(DrawablePtr pDrawable)
    * would fail.
    */
 
-  pBitmap = nxagentCreatePixmap(pDrawable -> pScreen, pDrawable -> width, pDrawable -> height, pDrawable -> depth);
+  pBitmap = nxagentCreatePixmap(pDrawable -> pScreen, pDrawable -> width, pDrawable -> height, pDrawable -> depth, CREATE_PIXMAP_USAGE_SCRATCH);
```
Comment (Ionic) [inline nx-X11/programs/Xserver/hw/nxagent/Drawable.c]: I've used `0` here. Given that the pixmap is saved for later use, marking it as scratch usage doesn't sound safe to me.

**Review by Ionic — CHANGES_REQUESTED** [review]
I'd like to use your backport. It'll mean I have to make a little bit more of an effort when rebasing, but you have split up the commits nicer and included references.

Don't forget to rebase.

File: nx-X11/programs/Xserver/hw/nxagent/Drawable.c (hunk 1 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/hw/nxagent/Drawable.c]: Ok. Fine by me. And, you are actually right.

**Comment by sunweaver** [comment]
@Ionic: about to push this PR, once #362 has been merged. Then, you can merge this one right away, too.

**Comment by sunweaver** [comment]
Ionic: ok. Branch updated. Next iteration for #410.

**Review by Ionic — APPROVED** [review]

**Pull request merged**